api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -442,9 +456,10 @@ private Types.NestedField findSourceColumn(String sourceName) {
     Builder identity(String sourceName, String targetName) {
       Types.NestedField sourceColumn = findSourceColumn(sourceName);
       checkAndAddPartitionName(targetName, sourceColumn.fieldId());
+      String normalizedTargetName = caseSensitive ? targetName : targetName.toLowerCase();
```

Comment:
I don't think this should normalize the target name. That could be passed by the caller and this should not modify it. I think it should continue to use the target name that was passed in.

The problem isn't with this method, it is with this version of the method, it is with the `identity(String sourceName)` version that assumes it can pass the source name as the target name. One solution is to lower case the string there, but I would prefer to preserve the case from the schema instead.

I think the easiest way to do that is to add a private identity method, like this:

```java
    Builder identity(String sourceName, String targetName) {
      return identity(findSourceColumn(sourceName), targetName);
    }

    Builder identity(String sourceName) {
      Types.NestedField sourceColumn = findSourceColumn(sourceName);
      return identity(sourceColumn, sourceColumn.name);
    }

    private Builder identity(String sourceColumn, String targetName) {
      checkAndAddPartitionName(targetName, sourceColumn.fieldId());
      PartitionField field =
          new PartitionField(
              sourceColumn.fieldId(), nextFieldId(), targetName, Transforms.identity());
      checkForRedundantPartitions(field);
      fields.add(field);
      return this;
    }
```

Reply:
I assume you meant this. The two salient differences are 

1. adding `public` to `Builder identity(String sourceName)`
2. changing `sourceColumn` type from `String` to `Types.NestedField` in `private Builder identity(String sourceColumn, String targetName))`


```
    Builder identity(String sourceName, String targetName) {
      return identity(findSourceColumn(sourceName), targetName);
    }

    public Builder identity(String sourceName) {
      Types.NestedField sourceColumn = findSourceColumn(sourceName);
      return identity(sourceColumn, sourceColumn.name());
    }

    private Builder identity(Types.NestedField sourceColumn, String targetName) {
      checkAndAddPartitionName(targetName, sourceColumn.fieldId());
      PartitionField field =
              new PartitionField(
                      sourceColumn.fieldId(), nextFieldId(), targetName, Transforms.identity());
      checkForRedundantPartitions(field);
      fields.add(field);
      return this;
    }
```

Reply:
Are you implying that the normalization of `targetName` in the other transform methods is OK to keep?

Reply:
Yes, I didn't intend to change the visibility of any of the methods. The new methods should all be private, though.

I think that this should be done for all of the configuration methods that create a partition name (`targetName`) instead of normalizing to lower case.

Reply:
I've reverted normalization of `targetName` in those methods too